cfy_manager/components/statusreporter/status_reporter.py

```diff
@@ -0,0 +1,69 @@
+#########
+# Copyright (c) 2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+from ...config import config
+from ...utils.common import sudo
+from ...logger import get_logger
+from ...utils.systemd import systemd
+from ..base_component import BaseComponent
+from ..components_constants import SOURCES
+from ...utils.install import yum_install, yum_remove
+
+STATUS_REPORTER = 'statusreporter'
+
+logger = get_logger('status-reporter')
+
+
+class StatusReporter(BaseComponent):
+    def __init__(self, skip_installation, reporter_type):
+        super(StatusReporter, self).__init__(skip_installation)
+        self.reporter_type = reporter_type
+
+    def _build_extra_config_flags(self):
+        return ''
+
+    def _should_install(self):
+        return True
```

Comment:
we already have the `skip_installation` check so instead of having this additionally, use that
in the reporters that need it (rabbitmq), do something like (in `__init__`): `self.skip_installation = self.skip_installation or not config[rabbitmq][join_cluster]`, or make it a `@property` in it